copier-update workflow lacks id-token: write — claude-code-action agent steps fail OIDC auth

## Summary

The `update` job in `.github/workflows/copier-update.yml.jinja` declares only `contents: write` and `pull-requests: write` permissions. The three `Agent Job *` steps invoke `anthropics/claude-code-action@v1`, which calls `setupGitHubToken()` → `getOidcToken()` to mint a Claude-app GitHub token. That OIDC call needs `id-token: write` on the job's `permissions:` block. Without it, the action retries 3× and aborts with `Error message: Unable to get ACTIONS_ID_TOKEN_REQUEST_URL env variable` / `Could not fetch an OIDC token`.

The agent steps are wrapped in `continue-on-error: true`, so the PR still opens — but with all three analysis sections rendered as "⚠️ Agent failed — see workflow log." and any conflict markers left unresolved.

## Evidence

Downstream `markdown-vault-mcp` ran the v1.5.1 → v1.6.0 update via this workflow:

- Run: https://github.com/pvliesdonk/markdown-vault-mcp/actions/runs/25634333290
- Resulting PR: https://github.com/pvliesdonk/markdown-vault-mcp/pull/434

Excerpt from the run log (Agent Job A — Conflict resolution):

```
Auto-detected mode: agent for event: workflow_dispatch
Requesting OIDC token...
Attempt 1 of 3...
Failed to get OIDC token: ...
error: Error message: Unable to get ACTIONS_ID_TOKEN_REQUEST_URL env variable
  at getOidcToken (src/github/token.ts:15:34)
  at async setupGitHubToken (src/github/token.ts:138:27)
  at async run (src/entrypoints/run.ts:164:27)
Attempt 1 failed: Could not fetch an OIDC token. Did you remember to add `id-token: write` to your workflow permissions?
... (retry 2, retry 3) ...
Operation failed after 3 attempts
##[error]Action failed with error: Could not fetch an OIDC token. Did you remember to add `id-token: write` to your workflow permissions?
```

The same failure occurs on every weekly run for every downstream consumer until fixed.

## Root cause

From `claude-code-action@v1` `src/github/token.ts:128-156` (`setupGitHubToken`):

1. If `inputs.github_token` is provided → use it.
2. Otherwise → mint OIDC token, exchange for app token.

The template does not pass `github_token`, so the action takes path (2). That path requires `id-token: write` at the job level (workflow-level fallback is not enough for OIDC).

## Fix

Add `id-token: write` to the `update` job's `permissions:` block in `.github/workflows/copier-update.yml.jinja`. Workflow-level `permissions: contents: read` stays as-is (it bounds `GITHUB_TOKEN` for any unprivileged jobs that get added later).

```yaml
jobs:
  update:
    runs-on: ubuntu-latest
    permissions:
      contents: write
      pull-requests: write
      id-token: write   # ← add this; required by claude-code-action's OIDC → Claude-app-token exchange
    steps:
      ...
```

## Verification

- Render the template against the smoke fixture; confirm the rendered `.github/workflows/copier-update.yml` contains `id-token: write`.
- Cut a patch release (v1.6.1) of the template; the next downstream `Copier Update` run should successfully complete the agent steps (or at least no longer fail on OIDC).

## Downstream impact

Every consumer of this template currently has a copier-update workflow that fails its agent enrichment steps on every run. Cleaner remediation lands automatically once each consumer pulls in the fixed template version via their next copier-update PR.

--- a/.github/workflows/copier-update.yml.jinja
+++ b/.github/workflows/copier-update.yml.jinja
@@ -235,8 +235,11 @@ jobs:
       # Each agent step constrains tool access via `claude_args
       # --allowed-tools` to enforce the spec's per-job tool boundaries
       # (Job A: file read/write + git commit on the working tree; Jobs B/C:
-      # read-only). Without this, a prompt-injection-vulnerable model could
-      # bypass the prose constraints in the prompt body.
+      # read on the working tree and the cloned template, plus a path-scoped
+      # write on their single output JSON at /tmp/agent-job-{b,c}.json).
+      # Without this, a prompt-injection-vulnerable model could bypass the
+      # prose constraints in the prompt body — e.g. attempt to write a
+      # workflow file or clobber another job's output JSON.
 
       - name: Pre-render Job A prompt
         id: prepare-prompt-a
@@ -305,7 +308,7 @@ jobs:
           claude_code_oauth_token: {{ '${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}' }}
           prompt: {{ '${{ steps.prepare-prompt-b.outputs.prompt }}' }}
           claude_args: |
-            --allowed-tools "Read,Bash(cat:*),Bash(git -C /tmp/template:*)"
+            --allowed-tools "Read,Write(//tmp/agent-job-b.json),Bash(cat:*),Bash(git -C /tmp/template:*)"
 
       - name: Pre-render Job C prompt
         id: prepare-prompt-c
@@ -339,7 +342,7 @@ jobs:
           claude_code_oauth_token: {{ '${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}' }}
           prompt: {{ '${{ steps.prepare-prompt-c.outputs.prompt }}' }}
           claude_args: |
-            --allowed-tools "Read,Bash(cat:*),Bash(git -C /tmp/template:*)"
+            --allowed-tools "Read,Write(//tmp/agent-job-c.json),Bash(cat:*),Bash(git -C /tmp/template:*)"
 
       - name: Ensure PR labels exist
         if: steps.changes.outputs.changed == 'true'

--- a/.github/workflows/copier-update.yml.jinja
+++ b/.github/workflows/copier-update.yml.jinja
@@ -235,11 +235,12 @@ jobs:
       # Each agent step constrains tool access via `claude_args
       # --allowed-tools` to enforce the spec's per-job tool boundaries
       # (Job A: file read/write + git commit on the working tree; Jobs B/C:
-      # read on the working tree and the cloned template, plus a path-scoped
-      # write on their single output JSON at /tmp/agent-job-{b,c}.json).
-      # Without this, a prompt-injection-vulnerable model could bypass the
-      # prose constraints in the prompt body — e.g. attempt to write a
-      # workflow file or clobber another job's output JSON.
+      # read everywhere, plus a bare Write — narrowed to the single output
+      # JSON at /tmp/agent-job-{b,c}.json by prose in each job's prompt).
+      # The path-scoped Write(//tmp/agent-job-b.json) form documented at
+      # code.claude.com/docs/en/permissions empirically doesn't match the
+      # agent's single-slash invocations under the SDK shipped with
+      # claude-code-action@v1, so the prose constraint is what's narrow.
 
       - name: Pre-render Job A prompt
         id: prepare-prompt-a
@@ -308,7 +309,7 @@ jobs:
           claude_code_oauth_token: {{ '${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}' }}
           prompt: {{ '${{ steps.prepare-prompt-b.outputs.prompt }}' }}
           claude_args: |
-            --allowed-tools "Read,Write(//tmp/agent-job-b.json),Bash(cat:*),Bash(git -C /tmp/template:*)"
+            --allowed-tools "Read,Write,Bash(cat:*),Bash(git -C /tmp/template:*)"
 
       - name: Pre-render Job C prompt
         id: prepare-prompt-c
@@ -342,7 +343,7 @@ jobs:
           claude_code_oauth_token: {{ '${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}' }}
           prompt: {{ '${{ steps.prepare-prompt-c.outputs.prompt }}' }}
           claude_args: |
-            --allowed-tools "Read,Write(//tmp/agent-job-c.json),Bash(cat:*),Bash(git -C /tmp/template:*)"
+            --allowed-tools "Read,Write,Bash(cat:*),Bash(git -C /tmp/template:*)"
 
       - name: Ensure PR labels exist
         if: steps.changes.outputs.changed == 'true'

--- a/.github/workflows/copier-update.yml.jinja
+++ b/.github/workflows/copier-update.yml.jinja
@@ -27,6 +27,11 @@ jobs:
     permissions:
       contents: write
       pull-requests: write
+      # Required by anthropics/claude-code-action@v1: setupGitHubToken() mints
+      # a Claude-app GitHub token via OIDC token exchange, which needs
+      # id-token: write at the job level. Without it the agent steps retry
+      # 3x then abort on "Could not fetch an OIDC token".
+      id-token: write
     steps:
       - name: Checkout
         uses: actions/checkout@v6

